src/cpp/fastdds/subscriber/DataReaderImpl.cpp

```diff
@@ -1637,6 +1639,8 @@ fastrtps::TopicAttributes DataReaderImpl::topic_attributes() const
 DataReaderListener* DataReaderImpl::get_listener_for(
         const StatusMask& status)
 {
+    std::lock_guard<std::mutex> _(listener_mutex_);
```

Comment:
Maybe it would be best to keep the final return clause outside the scope of this mutex.